unitex/src/fr/umlv/unitex/graphrendering/GraphBox.java

```diff
@@ -514,5 +514,128 @@ void drawOtherStandalone(Graphics2D g, DrawGraphParams params) {
 					old);
 		}
 	}
+
+  /**
+   * This method checks if the String s is valid or not.
+   *
+   * @param s the string to check
+   * @throws BackSlashAtEndOfLineException if the String ends with a backslash.
+   * @throws MissingGraphNameException if the String starts with a semicolon but is missing a graph name.
+   * @throws NoClosingQuoteException if the String contains a not closed quotation.
+   * @throws NoClosingSupException if the String contains a not closed chevron.
+   * @throws NoClosingRoundBracketException if the String contains a not closed brace.
+   */
+  public void checkString(String s)
```

Comment:
This function seems to be highly similar (96%) to [tokenizeTextWithoutTransduction](../../blob/9762cb327c5c04c9148abc413d69a579b53173ea/unitex/src/fr/umlv/unitex/graphrendering/GraphBox.java#L79-193). What was the motivation behind to duplicate the code? Do you see some possibility to refactor it to reduce duplication?